### ISSUE TITLE
Site Editor: change label for the Index template to "Default"

### ISSUE THREAD

**Issue by creativecoder** [WordPress/gutenberg#57324]
## What problem does this address?

The Index template is effectively a fallback, used when there are not more specific templates. This behavior makes the label "Index" unspecific and unhelpful without prior knowledge of the template hierarchy.

## What is your proposed solution?

Change the label of the template to "Default," so the name better matches the behavior.

**Comment by richtabor** [comment]
I don't think that "Default" is perhaps the right solution. I towards what is discussed in https://github.com/WordPress/gutenberg/issues/56244, to name it the home page.

**Comment by creativecoder** [comment]
Thanks for the link to that issue!

I think Index and Home are almost always different things--all of these would override index:

- front-page
- Home page set to latest posts
    - home.php
- Home page set to a selected page
    - Any specific template for that page (like a custom page template)
    - page.php
    - singular.php
   
My understanding is that index would be used for the home page only if none of those other templates exist. Based on this chart, it functions like a fallback, the default template if there is not a more specific template to use.

![image](https://github.com/WordPress/gutenberg/assets/1699996/484b81a5-8cfc-484e-8e21-ae848fa506c8)
